src/harness/harness.ts

```diff
@@ -1485,67 +1517,95 @@ namespace Harness {
             return;
 
             function checkBaseLines(isSymbolBaseLine: boolean) {
-                const fullBaseLine = generateBaseLine(fullResults, isSymbolBaseLine);
-
                 const fullExtension = isSymbolBaseLine ? ".symbols" : ".types";
-
                 // When calling this function from rwc-runner, the baselinePath will have no extension.
                 // As rwc test- file is stored in json which ".json" will get stripped off.
                 // When calling this function from compiler-runner, the baselinePath will then has either ".ts" or ".tsx" extension
                 const outputFileName = ts.endsWith(baselinePath, ts.Extension.Ts) || ts.endsWith(baselinePath, ts.Extension.Tsx) ?
-                    baselinePath.replace(/\.tsx?/, fullExtension) : baselinePath.concat(fullExtension);
-                Harness.Baseline.runBaseline(outputFileName, () => fullBaseLine, opts);
+                    baselinePath.replace(/\.tsx?/, "") : baselinePath;
+
+                if (!multifile) {
+                    const fullBaseLine = generateBaseLine(fullResults, isSymbolBaseLine);
+                    Harness.Baseline.runBaseline(outputFileName + fullExtension, () => fullBaseLine, opts);
+                }
+                else {
+                    Harness.Baseline.runMultifileBaseline(outputFileName, fullExtension, () => {
+                        return iterateBaseLine(fullResults, isSymbolBaseLine);
+                    }, opts);
+                }
             }
 
             function generateBaseLine(typeWriterResults: ts.Map<TypeWriterResult[]>, isSymbolBaseline: boolean): string {
-                const typeLines: string[] = [];
+                let result = "";
+                const gen = iterateBaseLine(typeWriterResults, isSymbolBaseline);
+                for (let {done, value} = gen.next(); !done; { done, value } = gen.next()) {
+                    const [, content] = value;
+                    result += content;
+                }
+                return result;
+            }
+
+            function *iterateBaseLine(typeWriterResults: ts.Map<TypeWriterResult[]>, isSymbolBaseline: boolean): IterableIterator<[string, string]> {
+                let typeLines = "";

```

Comment:
did you switch to `string` from `string[]` because string concatenation is fast enough in JS?

Reply:
No fast enough, but _faster_.